trie/zk_trie_impl.go

```diff
@@ -47,10 +48,13 @@ var (
 // ZkTrieImpl is the struct with the main elements of the ZkTrieImpl
 type ZkTrieImpl struct {
 	db        ZktrieDatabase
-	rootHash  *zkt.Hash
+	rootKey   *zkt.Hash
```

Comment:
> since this field is in fact the hash of "root node",

with this PR that is not necesarily true all the time. `rootKey` can be a temporariy key that we derive from `dirtyIndex`as well, which is just an incrementing counter.

Reply:
ok